### PR TITLE
Update koa-bodyparser to allow extra properties

### PR DESCRIPTION
**Pull request by getdaydream**
When I perform a post request using JSON , the rawBody should be sting, and body should be an object that using index signature to allow extra properties.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f21f990e9f8867ca328a7725cae68b0cf2fd06ad/types/koa-bodyparser/index.d.ts
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by anup-2s** [comment]
Hi @getdaydream, thanks for the contribution! I think I agree with the change to using `string` for `rawBody`, however, the change for `body` produces a lot of insecurity that you might want to look into using a type guard for, instead. For a bit more of a writeup on that, see the original PR that introduced the change: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26330

Definitely feel free to add comments if you have any more questions!

**Comment by RyanCavanaugh** [comment]
@anup-2s are you requesting changes on this, or approving as-is?

**Comment by anup-2s** [comment]
Ah, requesting changes, but was uncertain where to initiate that -- will start a review on the diff, sorry about that @RyanCavanaugh!

File: types/koa-bodyparser/index.d.ts [hunk 1]
```diff
@@ -18,8 +18,8 @@ import * as Koa from "koa";
 
 declare module "koa" {
     interface Request {
-        body: {} | null | undefined;
-        rawBody: {} | null | undefined;
+        body: { [x: string]: any } | null | undefined;
```
Comment (anup-2s) [inline types/koa-bodyparser/index.d.ts]: This change ends up producing a lot of type insecurity, and should probably be avoided. To accomplish this, consider using a typeguard or a runtime type validation library (or, simply cast `body` as an `any` to bypass). For more discussion, see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26330

**Review by anup-the-magic — CHANGES_REQUESTED** [review]

**Pull request closed without merging**